### ISSUE TITLE
Sweep_: format wandb run name in pytorch checkpoint

### ISSUE THREAD

**Issue by ayulockin** [ayulockin/pytorch-lightning#6]
### Description & Motivation

It would be great to format the wandb run name: 
`wandb.run.name`

into the checkpoint filename.

epoch: 10
loss: 0.5423

filename: "epoch{epoch}-loss{val_loss:.2f}" --> "epoch10-loss0.54"

filename: "{wandb.run.name}-{epoch}-{val_loss:.2f}" --> "swept_sky_54-epoch10-loss0.54"

I am not sure if this is already possible but I think the filename function in the ModelCallback can only access metrics so far. 
 # Suggested Solution: 
 In order to add the wandb run name into the checkpoint filename, you'll first need access to the wandb run object. By currently used design, access to this information is not part of the information flow in PyTorch Lightning, and therefore, cannot be retrieved from within a callback.

However, I could suggest a workaround you could use:

In your LightningModule define your training_step and validation_end like so:

```python
def training_step(self, batch, batch_nb):
    # ...
    self.log('train_loss', loss)

def validation_step(self, batch, batch_nb):
    # ...
    self.log('val_loss', loss)

# Save the weights during training.
def training_epoch_end(self, outputs):
    if self.current_epoch % 5 == 0:  # or whatever condition you like
        ckpt_name = f'{wandb.run.name}-epoch{self.current_epoch}-loss{self.trainer.logged_metrics["val_loss"]:.2f}.ckpt'
        self.trainer.save_checkpoint(ckpt_name)
```

Also, you can consider creating a custom callback, that uses the `on_train_epoch_end` hook for saving the model:

```python
class CustomModelCheckpoint(Callback):

    def __init__(self):
        super().__init__()

    def on_train_epoch_end(self, trainer, pl_module):
        if trainer.current_epoch % 5 == 0: # or whatever condition you prefer
            filename = f"{wandb.run.name}-epoch{trainer.current_epoch}-loss{trainer.logged_metrics['val_loss']:.2f}"
            trainer.save_checkpoint(f'{trainer.logger.log_dir}/{filename}.ckpt')
```

You then add this callback to your Trainer:

```python
trainer = Trainer(callbacks=[CustomModelCheckpoint()])
```

Note: `wandb.run.name` should be accessible at the point of saving the checkpoint assuming you are initializing wandb before Trainer.

This, however, is an ad-hoc fix, and currently, PyTorch Lightning does not have a built-in mechanism to be aware of wandb's run name in order to use into checkpoint names.

To enable this feature natively, changes would have to be made to the core pytorch-lightning code to include the Logger's name as part of the checkpoint name, and this decision would also need to consider logging services other than wandb. You can propose this feature request on the PyTorch Lightning GitHub repository.
